cql3/restrictions/statement_restrictions.cc

```diff
@@ -1266,7 +1266,15 @@ struct multi_column_range_accumulator {
             utils::chunked_vector<std::vector<managed_bytes_opt>> tuple_elems;
             if (tup.is_value()) {
                 tuple_elems = expr::get_list_of_tuples_elements(tup, *type_of(binop.rhs));
-            }
+            }   
+            for(size_t i = 0; i < tuple_elems.size(); ++i) {
+                for(size_t j = 0; j < lhs.elements.size(); ++j) {
+                    auto& col = expr::as<column_value>(lhs.elements.at(j));
+                    statements::request_validations::check_not_null(
+                            tuple_elems[i][j],
```

Comment:
Ok, I'm confused now... I know I asked you why you looped until j=lhs.elements.size() and not tuple_elems[i].size() - because you had tuple_elems[i][i], but now I realized you **also** have lhs.elements.at(j). What's going on - is lhs.elements guaranteed to be the same length as tuple_elems[i]? 

What happens if it's not the same length - do we get a good error message or the at(j) here will throw?

It seems there is a pre-existing issue on this case - https://github.com/scylladb/scylladb/issues/13241 - but let's sure we do not make it even worse. If it just leaves the same bug, it's fine - we don't have to solve all the bugs in one PR :-)

Reply:
So, you want me to change it back to lhs.elements.size(), right? If I'm wrong, please give me a hint :).

Reply:
Basically, the code seems to assume that lhs.elements.size() == tuple_elems[i].size() for each i - if it isn't, the code will fail in a weird way. Do we check that these two things are equal, earlier? If not, you can't assume they are equal.

Reply:
I tested with test.py, everything passed.

Reply:
I also tested this myself. The message is being printed when tuple size is lesser than lhs size. When tuple size is longer than lhs size, Scylla gives another error message like in #13241.